src/Util/util.jsx

```diff
@@ -43,6 +43,33 @@ export const countDirection = (city1, city2) => {
   }
 };
 
+/**
+ * Calculate precise azimuth (bearing) from city1 to city2
+ * Returns angle in degrees (0-360), where 0° is North, 90° is East, etc.
+ * Returns null if cities are at the same location
+ */
+export const calculateAzimuth = (city1, city2) => {
+  const latitudeDiff = city2.latitude - city1.latitude;
+  const longitudeDiff = city2.longitude - city1.longitude;
+
+  if (latitudeDiff === 0 && longitudeDiff === 0) {
+    return null;
+  }
+
+  // Calculate bearing using atan2
+  // atan2 gives angle from East (0°), counterclockwise
+  // We need to convert to bearing from North (0°), clockwise
+  const angleRad = Math.atan2(longitudeDiff, latitudeDiff);
```

Comment:
Use the Haversine formula for bearing calculation or leverage the existing geolib library.